### PR TITLE
Add feature of upload audience by file

### PR DESCRIPTION
**Pull request by moririnson**
- [ ] [The audiences property is now optional instead of required for a certain endpoint](https://developers.line.biz/en/reference/messaging-api/#create-upload-audience-group)
- [ ] [Create audience for uploading user IDs (by file)](https://developers.line.biz/en/reference/messaging-api/#create-upload-audience-group-by-file)
- [ ] [Add user IDs or Identifiers for Advertisers (IFAs) to an audience for uploading user IDs (by file](https://developers.line.biz/en/reference/messaging-api/#update-upload-audience-group-by-file)

### CODE REVIEW

File: src/LINEBot.php [hunk 1]
```diff
@@ -878,21 +879,51 @@ public function getNarrowcastProgress($requestId)
      */
     public function createAudienceGroupForUpdatingUserIds(
         $description,
-        $audiences,
+        $audiences = [],
```
Comment (moririnson) [inline src/LINEBot.php:882]: The audiences property has been changed from required to optional for the following endpoint.

File: src/LINEBot.php [hunk 2]
```diff
@@ -878,21 +879,51 @@ public function getNarrowcastProgress($requestId)
      */
     public function createAudienceGroupForUpdatingUserIds(
```
Comment (moririnson) [inline src/LINEBot.php:880]: I want to fix `updating` to `uploading`...

File: src/LINEBot.php [hunk 3]
```diff
@@ -878,21 +879,51 @@ public function getNarrowcastProgress($requestId)
...
+    public function createAudienceGroupForUploadingUserIdsByFile(
+        $description,
+        $filePath,
+        $isIfaAudience = false,
+        $uploadDescription = null
+    ) {
+        $params = [
+            'description' => $description,
+            'isIfaAudience' => $isIfaAudience,
+            'file' => new CURLFile($filePath, 'text/plain', 'file'),
```
Comment (moririnson) [inline src/LINEBot.php:917]: I think LINE Bot should not know CURL. But I could not think of a good idea. So I directly specify CURLFile.

**Review by Vaduz — APPROVED** [review]

File: src/LINEBot.php (hunk 2 above)
Comment (Vaduz) [reply src/LINEBot.php:880]: Rename the method and create alias for it. And warn duplication.

**Pull request merged**